### PR TITLE
rpcdaemon: use glaze in eth_getBlockByNumber

### PR DESCRIPTION
**Pull request by lupin012**

### CODE REVIEW

File: silkworm/silkrpc/json/block.cpp [hunk 1]
```diff
@@ -84,4 +79,319 @@ void to_json(nlohmann::json& json, const Block& b) {
...
+    struct glaze {
+        using T = GlazeJsonNullBlock;
+        static constexpr auto value = glz::object(
+            "jsonrpc", &T::jsonrpc,
+            "id", &T::id,
+            "result", &T::result);
+    };
+};
+
+void make_glaze_json_withdrawls_content(GlazeJsonBlock& block_json_data, const BlockBody& block) {
```
Comment (canepat) [inline silkworm/silkrpc/json/block.cpp:248]: `void make_glaze_json_withdrawals(const BlockBody& block, GlazeJsonBlock& json_block)`

File: silkworm/silkrpc/json/block.cpp [hunk 2]
```diff
@@ -84,4 +79,319 @@ void to_json(nlohmann::json& json, const Block& b) {
...
+}
+
+void make_glaze_json_null_content(std::string& reply, uint32_t id) {
+    GlazeJsonNullBlock block_json_data{};
+    block_json_data.id = id;
+
+    glz::write<glz::opts{.skip_null_members = false}>(block_json_data, reply);
+}
+
+void make_glaze_json_transaction_content(GlazeJsonTransaction& item, const silkworm::Transaction& transaction) {
```
Comment (canepat) [inline silkworm/silkrpc/json/block.cpp:269]: `void make_glaze_json_transaction(const silkworm::Transaction& tx, GlazeJsonTransaction& json_tx)`

File: silkworm/silkrpc/json/block.hpp [hunk 3]
```diff
@@ -24,4 +24,7 @@ namespace silkworm::rpc {
 
 void to_json(nlohmann::json& json, const Block& b);
 
+void make_glaze_json_content(std::string& reply, uint32_t id, const Block& b);
```
Comment (canepat) [inline silkworm/silkrpc/json/block.hpp:27]: `void make_glaze_json_content(uint32_t id, const Block& b, std::string& json_reply);`

File: silkworm/silkrpc/json/block.hpp [hunk 4]
```diff
@@ -24,4 +24,7 @@ namespace silkworm::rpc {
...
+void make_glaze_json_null_content(std::string& reply, uint32_t id);
```
Comment (canepat) [inline silkworm/silkrpc/json/block.hpp:28]: void make_glaze_json_null_content(uint32_t id, std::string& json_reply);

**Review by canepat — CHANGES_REQUESTED** [review]

File: silkworm/silkrpc/commands/eth_api.cpp [hunk 5]
```diff
@@ -252,18 +252,18 @@ Task<void> EthereumRpcApi::handle_eth_get_block_by_number(const nlohmann::json&
             ensure_post_condition(total_difficulty.has_value(), "no difficulty for block number=" + std::to_string(block_number));
             const Block extended_block{*block_with_hash, *total_difficulty, full_tx};
 
-            reply = make_json_content(request["id"], extended_block);
+            make_glaze_json_content(reply, request["id"], extended_block);
         } else {
-            reply = make_json_content(request["id"], {});
+            make_glaze_json_null_content(reply, request["id"]);
```
[the remark this answers is not in the document]
Comment (lupin012) [reply silkworm/silkrpc/commands/eth_api.cpp:257]: ok

File: silkworm/silkrpc/json/block.cpp [hunk 6]
```diff
@@ -84,4 +79,319 @@ void to_json(nlohmann::json& json, const Block& b) {
...
+    struct glaze {
+        using T = GlazeJsonBlock;
+        static constexpr auto value = glz::object(
+            "jsonrpc", &T::jsonrpc,
+            "id", &T::id,
+            "result", &T::result);
+    };
+};
+
+struct GlazeJsonNullBlock {
```
[the remark this answers is not in the document]
Comment (lupin012) [reply silkworm/silkrpc/json/block.cpp:234]: ok

File: silkworm/silkrpc/json/block.cpp (hunk 1 above)
Comment (lupin012) [reply silkworm/silkrpc/json/block.cpp:248]: ok

File: silkworm/silkrpc/json/block.cpp (hunk 2 above)
Comment (lupin012) [reply silkworm/silkrpc/json/block.cpp:269]: ok

File: silkworm/silkrpc/json/block.hpp (hunk 3 above)
Comment (lupin012) [reply silkworm/silkrpc/json/block.hpp:27]: ok

File: silkworm/silkrpc/json/block.hpp (hunk 4 above)
Comment (lupin012) [reply silkworm/silkrpc/json/block.hpp:28]: ok

**Review by canepat — APPROVED** [review]

**Pull request merged**